core/arch/arm/plat-hikey/main.c

```diff
@@ -93,3 +108,166 @@ void console_flush(void)
 {
 	pl011_flush(console_base());
 }
+
+static vaddr_t peripherals_base(paddr_t pa)
+{
+	void *va;
+
+	if (cpu_mmu_enabled()) {
+		va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
```

Comment:
`static` wouldn't work if the function is to be used to get different base addresses. It would just return the first one always.